### PR TITLE
make code compatible with mongo ruby driver >= 2.0.0

### PR DESCRIPTION
**Pull request by thikonom**
I made some changes to make the code compatible with the new version of the mongo  driver (>= 2.0.0).

### CODE REVIEW

File: lib/logstash/outputs/mongodb.rb [hunk 1]
```diff
@@ -62,11 +53,11 @@ def receive(event)
       if @generateId
         document['_id'] = BSON::ObjectId.new(nil, event["@timestamp"])
       end
-      @db.collection(event.sprintf(@collection)).insert(document)
+      @db[event.sprintf(@collection)].insert_one(document)
     rescue => e
       @logger.warn("Failed to send event to MongoDB", :event => event, :exception => e,
                    :backtrace => e.backtrace)
-      if e.error_code == 11000
+      if e.message =~ /^11000/
```
Comment (coreypobrien) [inline lib/logstash/outputs/mongodb.rb]: This needs to be /^E11000/

Here's a snippet of the error: `:exception=>#<Mongo::Error::OperationFailure: E11000 duplicate key error...`

**Comment by coreypobrien** [comment]
+1 after the fix for the duplicate key error
I upgraded my mongo driver which hit this bug and caused everything to stop working. This patch got everything back normal.

**Comment by thikonom** [comment]
@coreypobrien thanks for spotting the typo. I'm confused does it work for you or not ?

**Comment by coreypobrien** [comment]
Yeah it works for me except for the duplicate record error which just sends things into an infinite loop of retry hell :) Sorry about the confusing comment. I edited it to be clearer

**Comment by thikonom** [comment]
No worries mate. 
I amended the regex, should be fine now.

**Comment by coreypobrien** [comment]
This fixes #3 so we can keep using the latest mongo driver. There is also #8 which pegs the driver to an older version. I'd prefer this fix.

**Comment by purbon** [comment]
Thanks a lot for your contribution @thikonom. To be able to go forward with this PR it would be necessary that you sign the CLA (https://www.elastic.co/contributor-agreement).

Looking forward to get this in,
- purbon

**Comment by thikonom** [comment]
Hey @purbon, I just signed it.

**Comment by purbon** [comment]
thanks a lot @thikonom, I just merged this PR with #14 where  I will focus on bringing this changes all in including adding unit test to the plugin, so I will close this PR for now. Thanks a lot man! much appreciate your contribution.

**Pull request closed without merging**